[iPad] Mail content on icloud.com has incorrect size after leaving app and reopening

lilyspiniolas opened the pull request:
#### 89ae3701d721a85099dc2b741d899a5d0a77f0c4
<pre>
[iPad] Mail content on icloud.com has incorrect size after leaving app and reopening
<a href="https://bugs.webkit.org/show_bug.cgi?id=287110">https://bugs.webkit.org/show_bug.cgi?id=287110</a>
<a href="https://rdar.apple.com/131836301">rdar://131836301</a>

Reviewed by Wenson Hsieh and Abrar Rahman Protyasha.

On iPad, the icloud.com mail app may display the mail content with an incorrect
width after closing Safari and reopening it. This is caused by the window resize
events that fire during snapshotting after closing Safari, which puts the page
into a bad state. These changes extend the &apos;shouldSilenceWindowResizeEvents&apos;
quirk to apply to the icloud.com mail app on iOS, preventing the site
from acting upon window resize events during snapshotting.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleICloudQuirks):
(WebCore::Quirks::determineRelevantQuirks):

Canonical link: <a href="https://commits.webkit.org/289904@main">https://commits.webkit.org/289904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/acdffb052373c3eecc7a08296a8038745778de44

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88348 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7867 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42823 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93305 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39102 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90399 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16050 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68143 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25864 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91350 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6315 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79943 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48512 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6087 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/34353 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38210 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76469 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35241 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95148 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15523 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11375 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77014 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-type-on-non-root.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15779 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75799 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76264 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20655 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8560 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13808 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15539 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/20846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15280 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18729 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17062 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1830,7 +1830,25 @@ void Quirks::setTopDocumentURLForTesting(URL&& url)
 // FIXME(rdar://141554467): The set of static functions below will be generated from a JSON file in a future patch. For now, we just move the logic
-// for deciding if a particular quirk is needed to domain-specific functionss below:
+// for deciding if a particular quirk is needed to domain-specific functions below:
+#if PLATFORM(IOS) || PLATFORM(MAC)
```
Is this intentionally `PLATFORM(IOS)`-only? (i.e. excluding watchOS, visionOS, Catalyst)

whsieh reviewed (approved):
r=me, with a comment

lilyspiniolas committed:
[iPad] Mail content on icloud.com has incorrect size after leaving app and reopening
https://bugs.webkit.org/show_bug.cgi?id=287110
rdar://131836301

Reviewed by Wenson Hsieh and Abrar Rahman Protyasha.

On iPad, the icloud.com mail app may display the mail content with an incorrect
width after closing Safari and reopening it. This is caused by the window resize
events that fire during snapshotting after closing Safari, which puts the page
into a bad state. These changes extend the 'shouldSilenceWindowResizeEvents'
quirk to apply to the icloud.com mail app on iOS, preventing the site
from acting upon window resize events during snapshotting.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleICloudQuirks):
(WebCore::Quirks::determineRelevantQuirks):

Canonical link: https://commits.webkit.org/289904@main

whsieh commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -2044,6 +2044,24 @@ static void handleScriptToEvaluateBeforeRunningScriptFromURLQuirk(QuirksData& qu
 }
 #endif
 
+#if PLATFORM(IOS_FAMILY) || PLATFORM(MAC)
```
(For future reference — this could just be `PLATFORM(COCOA)`)
